sdk/python/kubeflow/trainer/hf_llm_training.py

```diff
@@ -1,70 +1,87 @@
     Trainer,
 )
-import torch
-from datasets import load_dataset
-from peft import LoraConfig, get_peft_model
-from urllib.parse import urlparse
-import os
-import json
+
+
+# Configure logger.
+log_formatter = logging.Formatter(
+    "%(asctime)s %(levelname)-8s %(message)s", "%Y-%m-%dT%H:%M:%SZ"
+)
+logger = logging.getLogger(__file__)
+console_handler = logging.StreamHandler()
+console_handler.setFormatter(log_formatter)
+logger.addHandler(console_handler)
+logger.setLevel(logging.INFO)
 
 
 def setup_model_and_tokenizer(model_uri, transformer_type, model_dir):
     # Set up the model and tokenizer
     parsed_uri = urlparse(model_uri)
     model_name = parsed_uri.netloc + parsed_uri.path
-    transformer_type_class = getattr(transformers, transformer_type)
 
-    model = transformer_type_class.from_pretrained(
+    model = transformer_type.from_pretrained(
         pretrained_model_name_or_path=model_name,
         cache_dir=model_dir,
         local_files_only=True,
-        device_map="auto",
         trust_remote_code=True,
     )
 
-    tokenizer = transformers.AutoTokenizer.from_pretrained(
+    tokenizer = AutoTokenizer.from_pretrained(
         pretrained_model_name_or_path=model_name,
         cache_dir=model_dir,
         local_files_only=True,
-        device_map="auto",
     )
 
-    tokenizer.pad_token = tokenizer.eos_token
-    tokenizer.add_pad_token = True
-
     # Freeze model parameters
     for param in model.parameters():
         param.requires_grad = False
 
     return model, tokenizer
 
 
-def load_and_preprocess_data(dataset_name, dataset_dir, transformer_type, tokenizer):
+def load_and_preprocess_data(dataset_dir, transformer_type, tokenizer):
     # Load and preprocess the dataset
-    print("loading dataset")
-    transformer_type_class = getattr(transformers, transformer_type)
-    if transformer_type_class != transformers.AutoModelForImageClassification:
-        dataset = load_dataset(dataset_name, cache_dir=dataset_dir).map(
-            lambda x: tokenizer(x["text"]), batched=True
+    logger.info("Load and preprocess dataset")
+
+    if transformer_type != AutoModelForImageClassification:
+        dataset = load_from_disk(dataset_dir)
+
+        logger.info(f"Dataset specification: {dataset}")
+        logger.info("-" * 40)
+
+        logger.info("Tokenize dataset")
+        # TODO (andreyvelich): Discuss how user should set the tokenizer function.
+        dataset = dataset.map(
+            lambda x: tokenizer(x["text"], padding="max_length", truncation=True),
+            batched=True,
         )
     else:
-        dataset = load_dataset(dataset_name, cache_dir=dataset_dir)
+        dataset = load_from_disk(dataset_dir)
 
-    train_data = dataset["train"]
+    # Check if dataset contains `train` key. Otherwise, load full dataset to train_data.
+    if "train" in dataset:
+        train_data = dataset["train"]
+    else:
+        train_data = dataset
 
     try:
         eval_data = dataset["eval"]
```

Comment:
Is it always "dataset["eval"]"

Reply:
@johnugeorge It depends on the dataset.
If dataset doesn't have eval data, we can use `dataset.train_test_split(test_size=0.1, stratify_by_column="label")`. In that case train and eval dataset will be store under `train` and `test` keys.
Should we think about various use-cases in the followup PRs @johnugeorge ?

Reply:
SGTM